src/corehost/cli/fxr_resolver.cpp

```diff
@@ -130,3 +130,13 @@ bool fxr_resolver::try_get_path(const pal::string_t& root_path, pal::string_t* o
 
     return true;
 }
+
+bool fxr_resolver::try_get_existing_fxr(pal::dll_t *out_fxr, pal::string_t *out_fxr_path)
+{
+    pal::dll_t fxr;
+    if (!pal::get_loaded_library(LIBFXR_NAME, "hostfxr_main", out_fxr, out_fxr_path))
```

Comment:
I suspect this will detect if hostfxr is loaded globally (eg. Windows), but will it work correctly x-plat?  I could see us saying that this is just a best effort, which I think is fine, as any process that ends up in this situation would be 'fine' on next launch.

Reply:
It works properly on Windows and Linux (at least the ones we run tests in the CI on), but apparently does not on OSX (which is why the OSX CI is failing on the test I added)

Reply:
Should also be working on OSX now.